Chummer/Backend/Uniques/Spell.cs

```diff
@@ -859,6 +874,42 @@ private IEnumerable<Improvement> RelevantImprovements(Func<Improvement, bool> fu
                 }
             }
         }
+
+        /// <summary>
+        /// Method to check we are only applying the highest focus to the spell dicepool
+        /// </summary>
+        private Improvement CompareFocusPower(Improvement objImprovement)
+        {
+            // get any bonded foci that add to the base magic stat and return the highest rated one's rating
+            IEnumerable<Focus> fociList =
+                _objCharacter.Foci.Where(x => x.GearObject.Bonus.InnerText == "MAGRating" && x.GearObject.Bonded);
+            int powerFocusRating = fociList.Any()
+                ? fociList.OrderByDescending(x => x.Rating)?.FirstOrDefault().Rating ?? default(int)
+                : default(int);
+            // If our focus is higher, add in a partial bonus
+            if (powerFocusRating > 0 && powerFocusRating < objImprovement.Value)
+            {
+                // This is hackz -- because we don't want to lose the original improvement's value
+                // we instantiate a fake version of the improvement that isn't saved to represent the diff
+                return new Improvement(_objCharacter)
```

Comment:
This is the weirdest part of this PR, I think--we're basically creating a fake improvement representing the diff. But we don't want to modify the real improvement's value, since if the power focus is ever unbonded, we'll end up wanting to put the full value improvement back into play. Open to suggestions!